### PR TITLE
 Allow printing a plan outside of a transaction

### PR DESCRIPTION
**Pull request by dain**
Clean up hack for generating plan for completion event with auto commit transactions.

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/sql/planner/DistributedExecutionPlanner.java [hunk 1]
```diff
@@ -114,10 +121,27 @@ private StageExecutionPlan doPlan(SubPlan root, Session session, ImmutableList.B
             dependencies.add(doPlan(childPlan, session, allSplitSources));
         }
 
+        // extract TableInfo
+        Map<PlanNodeId, TableInfo> tables = new HashMap<>();
+        root.getFragment().getRoot().accept(
```
Comment (kokosing) [inline presto-main/src/main/java/io/prestosql/sql/planner/DistributedExecutionPlanner.java]: Maybe you could use `io.prestosql.sql.planner.optimizations.PlanNodeSearcher#searchFrom(io.prestosql.sql.planner.plan.PlanNode, io.prestosql.sql.planner.iterative.Lookup)`?

```
searchFrom( root.getFragment().getRoot())
  .where(TableScanNode.class::isInstanceOf)
  .findAll()
  .stream()
  .collect(toImmutableMap(...));
```

**Comment by findepi** [comment]
@dain does `Add partial field to SortNode` fix https://github.com/prestodb/presto/issues/12087?

File: presto-main/src/main/java/io/prestosql/sql/planner/QueryPlanner.java [hunk 2]
```diff
@@ -883,7 +884,8 @@ private PlanBuilder sort(PlanBuilder subPlan, Optional<OrderBy> orderBy, List<Ex
                 new SortNode(
                         idAllocator.getNextId(),
                         subPlan.getRoot(),
-                        new OrderingScheme(orderBySymbols.build(), orderings)));
+                        new OrderingScheme(orderBySymbols.build(), orderings),
+                        isDistributedSortEnabled(session)));
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/sql/planner/QueryPlanner.java]: Split between partial/non partial sort happens in `io.prestosql.sql.planner.optimizations.AddExchanges.Rewriter#visitSort`

**Comment by dain** [comment]
> @dain does `Add partial field to SortNode` fix [prestodb/presto#12087](https://github.com/prestodb/presto/issues/12087)?

My PR retains the existing behavior, which is "any sort node is considered partial of the flag is on".  My goal is to eliminate session from here to prevent future uses of it in plan printing.

I can take a crack a moving this to AddExchanges if that is what you and @sopel39 want.

**Comment by sopel39** [comment]
> I can take a crack a moving this to AddExchanges if that is what you and @sopel39 want.

@dain it should be extremely easy. Partial exchange is created here: `io/prestosql/sql/planner/optimizations/AddExchanges.java:439`

**Comment by dain** [comment]
@sopel39 I updated to set partial in the right place.

@martint this is ready to review

**Pull request merged**